dev/update_python.py

```diff
@@ -0,0 +1,270 @@
+"""Script to update Python versions in the codebase."""
+
+import argparse
+import re
+from pathlib import Path
+
+
+def _compute_old_version(new_version):
+    """Compute the old version as the immediate previous minor version."""
+    major_str, minor_str = new_version.split(".")
+    major = int(major_str)
+    minor = int(minor_str)
+
+    if minor > 0:
+        old_minor = minor - 1
+        old_version = f"{major}.{old_minor}"
+    else:
+        raise ValueError("Minor version is 0, can't infer previous version.")
+    return old_version
+
+
+def _update_python_versions(
+    new_full_version,
+    patch_only=False,
+    dry_run=False,
+):
+    """Update Python version strings in the specified files."""
+    new_major_minor = ".".join(new_full_version.split(".")[:2])
+
+    if patch_only:
+        print(f"Updating patch version for {new_major_minor} to {new_full_version}")
+
+        # Define the version pattern to match any full version with the same major.minor
+        version_pattern = re.escape(new_major_minor) + r"\.\d+"
+
+        # Define the file patterns and corresponding replacement patterns
+        replacements = {
+            # Shell scripts
+            "dev/*.sh": [
+                # Update version in scripts
+                (
+                    r"(version=\$\{1:-)" + version_pattern + r"(\})",
+                    r"\g<1>" + new_full_version + r"\g<2>",
+                ),
+                # Update pyenv uninstall commands
+                (
+                    r"(pyenv uninstall -f flower-)" + version_pattern,
+                    r"\g<1>" + new_full_version,
+                ),
+            ],
+            # Python files
+            "**/*.py": [
+                # Update version assignments
+                (
+                    r'(["\'])' + version_pattern + r'(["\'])',
+                    r"\g<1>" + new_full_version + r"\g<2>",
+                ),
+            ],
+            # Documentation files
+            "doc/source/conf.py": [
+                # Update Python full version in conf.py
+                (
+                    r"(\.\.\s*\|python_full_version\|\s*replace::\s*)"
+                    + version_pattern,
+                    r"\g<1>" + new_full_version,
+                ),
+            ],
+            # ReStructuredText files
+            "doc/source/*.rst": [
+                # Update Python full version in rst files
+                (
+                    r"(`Python\s*"
+                    + re.escape(new_major_minor)
+                    + r"\s*<https://docs.python.org/"
+                    + re.escape(new_major_minor)
+                    + r"/>`_)",
+                    r"`Python "
+                    + new_full_version
+                    + " <https://docs.python.org/"
+                    + new_major_minor
+                    + "/>`_",
+                ),
+            ],
+            # PO files for localization
+            "doc/locales/*/LC_MESSAGES/framework-docs.po": [
```

Comment:
Those currently don't use the full version format, only `major.minor`. When I run it with `--patch-only` I get:

```diff
- Flower requires at least `Python 3.9 <https://docs.python.org/3.9/>`_, but `Python 3.10 <https://docs.python.org/3.10/>`_ or above is recommended.
+ Flower requires at least `Python 3.9.23 <https://docs.python.org/3.9/>`_, but `Python 3.10 <https://docs.python.org/3.10/>`_ or above is recommended.
```

Should we move them to the `else` branch or do we want to use the full version format?

Reply:
Yes they should indeed be removed, I'll push the fix